ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/keys/ContractKeysValidation.scala

```diff
@@ -66,27 +67,23 @@ private[transaction] object ContractKeysValidation {
       transactionEntry: DamlTransactionEntrySummary,
   ): StepResult[DamlTransactionEntrySummary] = {
     val keysValidationOutcome = transactionEntry.transaction
-      .foldInExecutionOrder[KeyValidationStatus](
-        Right(
-          KeyValidationState(activeStateKeys = contractKeyDamlStateKeys)
-        )
+      .foldInExecutionOrder[Either[KeyValidationError, List[KeyValidationState]]](
```

Comment:
Maybe a `type KeyValidationStatusStack` could improve readability further?

Reply:
I don’t mind changing it if you prefer but this type is literally only used here. I’d rather know the actual type I’m folding over than introduce another indirection. That said, your code your choice so if you want the synonym I’ll happily add it.

Reply:
I agree it's opinable; I'd rather jump and back than having to parse `]]]`, especially if the typedef tells me what I'm dealing with and so I'd have to do it probably only once in a while, but it's a very minor matter. Feel free to leave it as it is.

Reply:
Added a synonym in 0af2beaa7cd6c5925b20f59105c083f676a4db6f